source/s3_client.c

```diff
@@ -483,41 +528,67 @@ static struct aws_s3_vip *s_s3_client_vip_new(
     manager_options.bootstrap = client->client_bootstrap;
     manager_options.initial_window_size = SIZE_MAX;
     manager_options.socket_options = &socket_options;
-    manager_options.proxy_options = NULL;
     manager_options.host = aws_byte_cursor_from_string(vip->host_address);
     manager_options.max_connections = s_num_connections_per_vip;
     manager_options.shutdown_complete_callback = s_s3_client_vip_http_connection_manager_shutdown_callback;
     manager_options.shutdown_complete_user_data = client;
 
+    struct aws_uri proxy_uri;
+    AWS_ZERO_STRUCT(proxy_uri);
+    struct aws_http_proxy_options *proxy_options = NULL;
+    struct aws_tls_connection_options *proxy_tls_options = NULL;
     struct aws_tls_connection_options *manager_tls_options = NULL;
+
+    if (s_s3_client_get_proxy_uri(client, &proxy_uri) == AWS_OP_SUCCESS) {
+        proxy_options = aws_mem_calloc(client->allocator, 1, sizeof(struct aws_http_proxy_options));
+        proxy_options->host = proxy_uri.host_name;
+        proxy_options->port = proxy_uri.port;
+
+        manager_options.proxy_options = proxy_options;
+    }
+
     if (client->tls_connection_options != NULL) {
         manager_tls_options = aws_mem_calloc(client->allocator, 1, sizeof(struct aws_tls_connection_options));
-        if (manager_tls_options == NULL) {
```

Comment:
Taking this out because the client has so far been written with the assumption that we're adding the OOM exception.